contracts/utils/KlimaCarbonRetirements.sol

```diff
@@ -0,0 +1,287 @@
+     */
+    modifier onlyOwner() {
+        require(owner() == _msgSender(), "Ownable: caller is not the owner");
+        _;
+    }
+
+    /**
+     * @dev Leaves the contract without owner. It will not be possible to call
+     * `onlyOwner` functions anymore. Can only be called by the current owner.
+     *
+     * NOTE: Renouncing ownership will leave the contract without an owner,
+     * thereby removing any functionality that is only available to the owner.
+     */
+    function renounceOwnership() public virtual onlyOwner {
+        _transferOwnership(address(0));
+    }
+
+    /**
+     * @dev Transfers ownership of the contract to a new account (`newOwner`).
+     * Can only be called by the current owner.
+     */
+    function transferOwnership(address newOwner) public virtual onlyOwner {
+        require(
+            newOwner != address(0),
+            "Ownable: new owner is the zero address"
+        );
+        _transferOwnership(newOwner);
+    }
+
+    /**
+     * @dev Transfers ownership of the contract to a new account (`newOwner`).
+     * Internal function without access restriction.
+     */
+    function _transferOwnership(address newOwner) internal virtual {
+        address oldOwner = _owner;
+        _owner = newOwner;
+        emit OwnershipTransferred(oldOwner, newOwner);
+    }
+}
+
+// File: contracts/utils/Ownable.sol
+
+// File: contracts/KlimaCarbonRetirements.sol
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+
+/**
+    @title Klima Retirement Storage
+    @notice This is used to store any offset retirements made through Klima retirement helper contracts.
+ */
+contract KlimaCarbonRetirements is Ownable {
+    struct Retirement {
+        uint256 totalRetirements;
+        uint256 totalCarbonRetired;
+        uint256 totalClaimedNFT;
+        mapping(uint256 => address) retiredPool;
+        mapping(uint256 => uint256) retiredAmount;
+        mapping(uint256 => string) retirementBeneficiary;
+        mapping(address => uint256) totalPoolRetired;
+    }
+
+    mapping(address => Retirement) public retirements;
+    mapping(address => bool) public isHelperContract;
+    mapping(address => bool) public isMinterContract;
+
+    uint256 totalCarbonRetired;
+
+    /**
+        @notice Stores the details of an offset transaction for future use
+        @param _retiree Address of the retiree. Not the address of a helper contract.
+        @param _pool Address of the carbon pool token.
+        @param _amount Number of tons offset. Expected is with 18 decimals.
+     */
+    function carbonRetired(
+        address _retiree,
+        address _pool,
+        uint256 _amount
+    ) public returns (bool) {
+        require(
+            isHelperContract[msg.sender],
+            "Caller is not a defined helper contract"
+        );
+
+        Retirement storage info = retirements[_retiree];
+
+        info.retiredPool[info.totalRetirements] = _pool;
```

Comment:
`info.totalRetirements` can be read once into memory to avoid the second SLOAD:
```js
uint256 totalRetirements = info.totalRetirements;
info.retiredPool[totalRetirements] = _pool;
info.retiredAmount[totalRetirements] = _amount;
```